File: src/main/java/io/openfuture/openmessanger/service/UserService.java
package io.openfuture.openmessanger.service;

import java.util.Collection;

import io.openfuture.openmessanger.repository.entity.User;
import io.openfuture.openmessanger.web.request.user.UserDetailsRequest;
import io.openfuture.openmessanger.web.response.UserDetailsResponse;

public interface UserService {
    Collection<User> getAllRecipientsBySender(String username);

    Collection<User> getAllUsers();

    User getByEmail(final String email);

    UserDetailsResponse getUserDetails(UserDetailsRequest request);
}

File: src/main/kotlin/io/openfuture/openmessanger/service/UserService.kt
package io.openfuture.openmessanger.service

import io.openfuture.openmessanger.repository.entity.User
import io.openfuture.openmessanger.web.request.user.UserDetailsRequest
import io.openfuture.openmessanger.web.response.UserDetailsResponse

interface UserService {
    fun getAllRecipientsBySender(username: String?): Collection<User?>?
    fun allUsers(): Collection<User?>
    fun getByEmail(email: String?): User?
    fun getUserDetails(request: UserDetailsRequest): UserDetailsResponse
}